src/components/Convert/routes/psm.ts

```diff
@@ -0,0 +1,80 @@
+import { BigNumber, ethers } from "ethers";
+import { HlpConfig, Network } from "../../..";
+import { PSM_GAS_LIMIT } from "../../../config/hlp";
+import { HPSM__factory } from "../../../contracts";
+import { ConvertQuoteInput, ConvertTransactionInput, Quote } from "../Convert";
+import { isTokenPegged } from "../isTokenPegged";
+import { PSM_WEIGHT, WeightInput } from "./weights";
+
+let transactionFeeCache: Record<Network, BigNumber | null> = {
+  arbitrum: null,
+  ethereum: null,
+  polygon: null
+};
+
+const TRANSACTION_FEE_UNIT = ethers.utils.parseEther("1");
+
+export const psmWeight = async (input: WeightInput) => {
+  const { fromToken, toToken, provider, network } = input;
+
+  if (!provider) return 0;
+
+  const isWithdraw = await isTokenPegged(fromToken.address, toToken.address, provider, network);
+  const isDeposit = await isTokenPegged(toToken.address, fromToken.address, provider, network);
```

Comment:
```suggestion
  const [isWithdraw, isDeposit] = await Promise.all([
    isTokenPegged(fromToken.address, toToken.address, provider, network),
    isTokenPegged(toToken.address, fromToken.address, provider, network)
  ]);
```
Please make sure to do that whenever you have a sequence of `await` where the lines don't depend on each other

Reply:
done